### ISSUE TITLE
Linter is missing non existing enums in the default

### ISSUE THREAD

**Issue by savage-alex** [stoplightio/spectral#1402]
Spectral 5.6.0

Given the following schema property
```yaml
property:
          type: string
          enum: [s, b, a, c]
          description: 'example schema for bug report'
          default: ''
          example: 'a'
```

Note that the validation on examples works great but no validation happens on default so I can leave it empty or put anything inside without spectral noticing.

**Comment by P0lip** [comment]
Hey!
I'm on it. I'm not sure yet whether a new rule will be introduced for this one or not, but regardless of the choice, Spectral will be able to validate some of the patterns you mentioned.

**Comment by P0lip** [comment]
The PR is up.
